x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/analysis/VerifierTests.java

```diff
@@ -58,8 +81,280 @@ public void testMisspelledColumnWithMultipleOptions() {
+                error(idxr, "foo where ip_range_field == ''"));
+    }
+
+    public void testObject() {
+        final IndexResolution idxr = loadIndexResolution("mapping-object.json");
+        accept(idxr, "foo where endgame.pid == 0");
+
+        assertEquals("1:11: Unknown column [endgame.pi], did you mean [endgame.pid]?",
+                error(idxr, "foo where endgame.pi == 0"));
+
+    }
+
+    public void testNested() {
+        final IndexResolution idxr = loadIndexResolution("mapping-nested.json");
+        accept(idxr, "foo where processes.pid == 0");
+
+        assertEquals("1:11: Unknown column [processe.pid], did you mean any of [processes.pid, processes.path, processes.path.keyword]?",
+                error(idxr, "foo where processe.pid == 0"));
+    }
+
+    public void testGeo() {
+        final IndexResolution idxr = loadIndexResolution("mapping-geo.json");
+        assertEquals("1:11: Cannot use field [location] with unsupported type [geo_point]",
+                error(idxr, "foo where location == 0"));
+        assertEquals("1:11: Cannot use field [site] with unsupported type [geo_shape]",
+                error(idxr, "foo where site == 0"));
+    }
+
+    public void testIP() {
+        final IndexResolution idxr = loadIndexResolution("mapping-ip.json");
+        accept(idxr, "foo where ip_addr == 0");
+    }
+
+    public void testJoin() {
+        final IndexResolution idxr = loadIndexResolution("mapping-join.json");
+        accept(idxr, "foo where serial_event_id == 0");
+    }
+
+    public void testMultiField() {
+        final IndexResolution idxr = loadIndexResolution("mapping-multi-field.json");
+        accept(idxr, "foo where multi_field.raw == 'bar'");
+
+        assertEquals("1:11: [multi_field.english == 'bar'] cannot operate on first argument field of data type [text]: " +
+                        "No keyword/multi-field defined exact matches for [english]; define one or use MATCH/QUERY instead",
+                error(idxr, "foo where multi_field.english == 'bar'"));
+
+        accept(idxr, "foo where multi_field_options.raw == 'bar'");
+        accept(idxr, "foo where multi_field_options.key == 'bar'");
+
+        accept(idxr, "foo where multi_field_ambiguous.one == 'bar'");
+        accept(idxr, "foo where multi_field_ambiguous.two == 'bar'");
+        assertEquals("1:11: [multi_field_ambiguous.normalized == 'bar'] cannot operate on first argument field of data type [keyword]: " +
+                        "Normalized keyword field cannot be used for exact match operations",
+                error(idxr, "foo where multi_field_ambiguous.normalized == 'bar'"));
+
+        assertEquals("1:11: [multi_field_nested.dep_name == 'bar'] cannot operate on first argument field of data type [text]: " +
+                        "No keyword/multi-field defined exact matches for [dep_name]; define one or use MATCH/QUERY instead",
```

Comment:
Same here about MATCH/QUERY predicates.

Reply:
Discussed with Costin, the behavior is correct, the error message can be improved in the context of EQL. Agreed  on leaving the test as is and open an "issue" for error messaging improvement for this particular case. Possibly can bring up for discussion for our weekly.